solr/core/src/java/org/apache/solr/update/VersionBucket.java

```diff
@@ -31,14 +31,41 @@
  * ignores the <code>lockTimeoutMs</code>.
  */
 public class VersionBucket {
-  public long highest;
 
+  private long highest;
+
+  /**
+   * @param highest the initial value of this bucket highest version.
+   */
+  public VersionBucket(long highest) {
+    setHighestIfGreater(highest);
```

Comment:
if I am reading the code correctly, this will be a noop because the `setHighestIfGreater ` method checks `if (highest != 0)` which coming from the constructor will always be false.